### PR TITLE
Make sure the JAXBContext is initialized at build time when building native

### LINKED ISSUE #2205: [camel] loading xml routes jvm vs native
I've noticed that loading routes in jvm and native mode does not behave the same, in particular the native mode seems not to handle xmlns & co.

Load in native mode:
```xml
<routes>
    <route>
        <from uri="netty4-http:0.0.0.0:8080"/>
        <convertBodyTo type="java.lang.String"/>
        <log message="${body}"/>
    </route>
</routes>
```

Does not load in native mode:
```xml
<routes xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns="http://camel.apache.org/schema/spring" 
    xsi:schemaLocation=" http://camel.apache.org/schema/spring http://camel.apache.org/schema/spring/camel-spring.xsd">
    <route>
        <from uri="netty4-http:0.0.0.0:8080"/>
        <convertBodyTo type="java.lang.String"/>
        <log message="${body}"/>
    </route>
</routes>

```

### PR DESCRIPTION
**Pull request by gnodet**
Fixes #2205

### CODE REVIEW

**Comment by gnodet** [comment]
This also relates to #2417

File: extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java
Comment (gsmet) [inline extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java]: Mmmmh?

**Review by gsmet — CHANGES_REQUESTED** [review]
Wondering why you commented the `RuntimeInitializedClassBuildItem` addition? Should it be removed? Is it an error?

**Comment by oscerd** [comment]
Can we get this one merged, This would fix the integration tests for camel-*

**Comment by gsmet** [comment]
@oscerd as mentioned above, there is some code commented out and I don't know if it's a mistake or code we should remove.

I won't merge it until I get a proper answer.

**Comment by oscerd** [comment]
Yes, just noticed that sorry @gsmet

**Comment by oscerd** [comment]
I'm testing the effect of this one btw

**Comment by oscerd** [comment]
@gsmet I can confirm this fix the integration tests, lets wait for @gnodet and sorry for the noise

**Comment by oscerd** [comment]
And btw it would be better to completely remove that stuff, without comment as you said.

**Comment by gsmet** [comment]
@oscerd it might make sense for Camel but JAXB is consumed by other extensions and I would like to understand the rationale of this change before merging it.

**Comment by oscerd** [comment]
I tested different integration tests even not camel related. Just for completeness :-)

File: extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java
Comment (gnodet) [reply extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java]: My understanding is that the JAXB `com.sun.xml.internal.bind.v2.runtime.reflect.opt.Injector` class is only used from `com.sun.xml.internal.bind.v2.runtime.reflect.opt.AccessorInjector#prepare` method, which is substituted to an empty method (see https://github.com/quarkusio/quarkus/blob/master/extensions/jaxb/runtime/src/main/java/io/quarkus/jaxb/runtime/graal/JAXBSubstitutions.java#L83).  The result is that there is no reference to this class anymore and GraalVM throws an exception when building compiling to native.
I suppose this may have been caused by a change in Graal.

Comment (gnodet) [reply extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java]: @gsmet I added another commit to remove the commented code

Comment (gsmet) [reply extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java]: @gnodet cool thanks. Sorry to be picky but could you amend the commit message with your explanation. That would be useful for history if one day we ask ourselves why we removed it.

Once it's done, I'll merge.

Comment (gnodet) [reply extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java]: @gsmet done

Comment (oscerd) [reply extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java]: this one is really important to make the camel extension works.

**Review by gsmet — APPROVED** [review]

**Pull request merged**
